packages/backend-plugin-api/src/wiring/createBackendModule.ts

```diff
@@ -55,6 +56,17 @@ export function createBackendModule(
   options: CreateBackendModuleOptions,
 ): BackendFeature {
   function getRegistrations() {
+    if (!ID_PATTERN.test(options.moduleId)) {
+      console.warn(
+        `WARNING: The moduleId '${options.moduleId}' for plugin '${options.pluginId}', will be invalid soon please must match the pattern ${ID_PATTERN} (letters, digits, and dashes only, starting with a letter)`,
+      );
+    }
+    if (!ID_PATTERN_OLD.test(options.moduleId)) {
+      throw new Error(
+        `Invalid moduleId '${options.moduleId}' for plugin '${options.pluginId}', must match the pattern ${ID_PATTERN} (letters, digits, and dashes only, starting with a letter)`,
+      );
+    }
```

Comment:
@Rugvip would something like this work for having a warning before imposing the new pattern? note sure if we have something else that the console.warn on this level

The ID_PATTERN_OLD is currently failing